### ISSUE TITLE
user assertions migration generating invalid data

### ISSUE THREAD

**Issue by brucehyslop** [AtlasOfLivingAustralia/la-pipelines#291]
The migration of userassertions from biocache cassanda `occ.qa` to `biocache.userassertions` constructs a single JSON array of all the user assertions for a occurrence record.

This process is generating invalid JSON.
see: https://aws-biocache-quoll.ala.org.au/ws/occurrences/e057144a-9927-47da-9750-5fedf78b22f5/assertions

**Comment by brucehyslop** [comment]
the issue appears to be that the `snapshot` value has an extra un-escaped quote `"` character added to the end.

**Comment by brucehyslop** [comment]
the first quote from the `snapshot` value for user assertion uuid `3913b5f2-fe2f-4bf7-85f8-95daadb416f8` is also missing. 

`"snapshot": "[{\rowKey\":\"dr366|OPP523627-40\", ...`

**Comment by djtfmartin** [comment]
Assume this is all related to #219.

I'll truncate the user assertion table for now in the dev environment as this is breaking a lot of the services.

**Comment by javier-molina** [comment]
Migration script is here https://github.com/AtlasOfLivingAustralia/biocache-service/tree/epic/pipelines/develop/scripts/migrateUserAssertions

**Comment by alexhuang091** [comment]
This is the data exported from Cassandra by running `cqlsh -e "copy occ.qa TO '/data/tmp/qa_dump.cql' with DELIMITER='\t' and header=true;";`

![1](https://user-images.githubusercontent.com/61677987/115488405-c4c8ef00-a29d-11eb-8c48-a67b2d8f2e60.jpg)

Look at the start/end of snapshot. I'll go to Cassandra to see if the original data is wrong.

**Comment by alexhuang091** [comment]
This is the data exported by running `cqlsh -e "copy occ.qa TO '/data/tmp/qa_dump.cql' with DELIMITER='\t' and header=true;";`

![1](https://user-images.githubusercontent.com/61677987/115488840-88e25980-a29e-11eb-8eff-18f3aafe91cc.jpg)

The first snapshot has no "" around it, but the second has. I'll check if it's a problem.

**Comment by alexhuang091** [comment]
And there are data like `\"userQualityAssertion\":\"[{\\\"uuid\\\":`

**Comment by alexhuang091** [comment]
As I show in previous picture, `cqlsh COPY TO` sometimes adds surrounding `""` on a value. It not only happens on `snapshot` field, I also saw it happen on `comment` field.

To make things worse, the original CSV reader we used in the script can't read correctly in this case:
See the leading `"` in snapshot field of 2nd record just disappeared

![1](https://user-images.githubusercontent.com/61677987/115657259-ed261b80-a379-11eb-8a96-c28e721634c9.jpg)


So my solution is:

- Remove any surrounding `""` on a field
- Use normal file reading plus string split to get fields.
- CSV writing we originally used also have some issue, I used normal file reading instead.


The result: I ran old/new script on the same data file and both generated 15 records:
- 5 records generated by old script has JSON parsing error
- 0 record by new script has JSON parsing error.

It might need more testing on test data set.
